### PR TITLE
updated file system + seeding

### PR DESCRIPTION
**Pull request by Ragansu**
I have updated the file address in few older scripts and added a new cpp file for keeping a random seed register. 

Regards 
RC

### CODE REVIEW

File: scripts/ramdom_seed_register.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        if (usedSeed == seed)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void writeRamdomSeedToFile(const std::vector<int> &generateSeedList, const std::string &filename)
```
Comment (cjh1) [inline scripts/ramdom_seed_register.cpp]: typo Ramdon => Random, same else where and in the filename.

File: scripts/ramdom_seed_register.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    std::string current_file = "/global/cfs/cdirs/m4287/hep/genHEPdata/scripts/";
+    int seed;
+    std::vector<int> generateSeedList;
+
+    int seed_num = 0;
+    while (seed_num < max_seed)
+    {
+        seed = generateSeed();
+        if (isSeedUsed(seed, filename))
```
Comment (cjh1) [inline scripts/ramdom_seed_register.cpp]: Re-reading this file from disk every time is going to pretty inefficient. Better to read the seeds outside the loop into a set or vector, then you can just check without hitting disk.

**Review by cjh1 — COMMENTED** [review]
A couple of inline comments. I would also stick with a single file end for C++ files, so either `.C` or `.cpp`. Similarly it seems the existing C++ file using camelCase rather than snake_case I would stick with one for consistence.

File: scripts/ramdom_seed_register.cpp (hunk 1 above)
Comment (Ragansu) [reply scripts/ramdom_seed_register.cpp]: Oh my bad. I will fix and push it to the PR

File: scripts/ramdom_seed_register.cpp (hunk 2 above)
Comment (Ragansu) [reply scripts/ramdom_seed_register.cpp]: I see. That explains the delay. I noticed when testing. I will replace it with a vector so it stays in RAM.

File: scripts/ramdom_seed_register.cpp (hunk 1 above)
Comment (Ragansu) [reply scripts/ramdom_seed_register.cpp]: I prefer cpp. I was not sure if root works with cpp file extension. I will change it and see how it goes.

**Comment by Ragansu** [comment]
I have fixed the issued mentioned before and indeed its faster.

**Comment by Ragansu** [comment]
I havent fixed the case yet. I will review the old code process.C and see if I can bring it all to one convention.

**Comment by cjh1** [comment]
> I havent fixed the case yet. I will review the old code process.C and see if I can bring it all to one convention.

Lets just rename `random_seed_register.cpp` to `randomSeedRegister.cpp` so its consistent with the existing file `preProcess.cpp`

**Comment by Ragansu** [comment]
Yes let's go it. Great call

**Comment by Ragansu** [comment]
Can you please do it. I don't have my laptop at the moment.

**Comment by cjh1** [comment]
> Can you please do it. I don't have my laptop at the moment.

Sure

**Review by Ragansu — COMMENTED** [review]
Looks good

File: scripts/randomSeedRegister.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            std::cout << "Seed is already used." << std::endl;
+            continue;
+        }
+        std::cout << "Generated seed: " << seed << std::endl;
+        auto seed_str = std::to_string(seed);
+
+        auto command = std::string("sbatch ") + current_file + "run_batch.sh " + seed_str + " " + process + " " + label;
+
+        system(command.c_str());
```
Comment (cjh1) [inline scripts/randomSeedRegister.cpp:81]: We should probably do some error checking here? I believe on POSIX platforms it should return the exit code of the command.

Comment (Ragansu) [reply scripts/randomSeedRegister.cpp:81]: I am not very familiar with raising errors. Can you add that. I will check it tomorrow.

**Review by cjh1 — APPROVED** [review]

**Pull request merged**